src/appleseed/foundation/math/beziercurve.h

```diff
@@ -116,15 +125,20 @@ class BezierCurve1
     typedef typename Base::VectorType VectorType;
     typedef typename Base::AABBType AABBType;
     typedef typename Base::MatrixType MatrixType;
+    typedef typename Base::ColorType ColorType;
 
     // Constructors.
     BezierCurve1();
-    BezierCurve1(const VectorType ctrl_pts[2], const ValueType width);
-    BezierCurve1(const VectorType ctrl_pts[2], const ValueType width[2]);
-    BezierCurve1(const BezierCurve1& curve, const MatrixType& xfm);
+    BezierCurve1(const VectorType ctrl_pts[2], const ValueType width,
+                 const ValueType opacity, const ColorType color);
+    BezierCurve1(const VectorType ctrl_pts[2], const ValueType width[2],
+                 const ValueType opacity[2], const ColorType color[2]);
+    BezierCurve1(const BezierCurve1& curve, const MatrixType& xfm, bool basis = false);
```

Comment:
What does `basis` mean? Maybe you can find a better name? Also document its meaning.